src/main/java/io/netty/incubator/codec/http3/Http3FrameCodec.java

```diff
@@ -721,6 +723,69 @@ private void unsetState(int toUnset) {
         }
     }
 
+    private static final class WriteResumptionListener
+            implements GenericFutureListener<Future<? super QuicStreamChannel>> {
+        private static final Object FLUSH = new Object();
+        private final Deque<Object> buffer;
+        private final ChannelHandlerContext ctx;
+        private final Http3FrameCodec codec;
+
+        WriteResumptionListener(ChannelHandlerContext ctx, Http3FrameCodec codec) {
+            this.ctx = ctx;
+            this.codec = codec;
+            buffer = new ArrayDeque<>(4); // assuming we will buffer header, data, trailer and a flush
+            assert codec.qpackAttributes != null;
+            codec.qpackAttributes.whenEncoderStreamAvailable(this);
```

Comment:
nit: calling `this` in a constructor is quite error-prone as you may pass an instance which is not "fully constructed yet". Better pattern is to add a static factory method, construct the object first and then use the new instance. something like:

```java
static WriteResumptionListener  newInstance(ChannelHandlerContext ctx, Http3FrameCodec codec) {
    WriteResumptionListener listener = new WriteResumptionListener(ctx, codec);
    codec.qpackAttributes.whenEncoderStreamAvailable(listener);
    return listener;
}
```